### PR TITLE
fix(upload): prevent form submit during file upload

### PR DESCRIPTION
**Pull request by stonebuzz**
If the user sends a file to GLPI and then tries to submit the form, 

The added element may be truncated (especially images) 

Nothing prevents the user from adding the element, even if the file has not been fully uploaded.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !28447

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: src/Html.php [hunk 1]
```diff
@@ -5577,6 +5577,8 @@ class='form-control'
             maxFileSize: {$max_file_size},
             maxChunkSize: {$max_chunk_size},
             add: function (e, data) {
+               // disable submit button during upload
+               $(this).closest('form').find('button:submit').prop('disabled', 'disabled');
```
Comment (cedric-anne) [inline src/Html.php:5581]: 1. Using `:submit` will also disable `<input type="submit" />` elements (can be used is specific forms).
2. `$().prop()` value parameter should be a boolean, according to documentation.
```suggestion
               $(this).closest('form').find(':submit').prop('disabled', true);
```

File: src/Html.php [hunk 2]
```diff
@@ -5592,16 +5594,22 @@ class='form-control'
                   $('#{$p['filecontainer']}'),
                   '{$p['editor_id']}'
                );
+               // enable submit button after upload
+               $(this).closest('form').find('button:submit').prop('disabled', '');
```
Comment (cedric-anne) [inline src/Html.php:5598]:
```suggestion
               $(this).closest('form').find(':submit').prop('disabled', false);
```

File: src/Html.php [hunk 3]
```diff
@@ -5592,16 +5594,22 @@ class='form-control'
...
                // remove required
                 $('#fileupload{$p['rand']}').removeAttr('required');
             },
             fail: function (e, data) {
+                // enable submit button after upload
+                $(this).closest('form').find('button:submit').prop('disabled', '');
```
Comment (cedric-anne) [inline src/Html.php:5604]:
```suggestion
                $(this).closest('form').find(':submit').prop('disabled', false);
```

File: src/Html.php [hunk 4]
```diff
@@ -5592,16 +5594,22 @@ class='form-control'
...
                const err = 'responseText' in data.jqXHR && data.jqXHR.responseText.length > 0
                   ? data.jqXHR.responseText
                   : data.jqXHR.statusText;
                alert(err);
             },
             processfail: function (e, data) {
+                // enable submit button after upload
+                $(this).closest('form').find('button:submit').prop('disabled', '');
```
Comment (cedric-anne) [inline src/Html.php:5612]:
```suggestion
                $(this).closest('form').find(':submit').prop('disabled', false);
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**